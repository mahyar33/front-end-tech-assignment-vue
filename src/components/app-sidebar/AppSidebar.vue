<template>
  <aside class="app-sidebar">
    <ul class="app-sidebar__menu">
      <app-sidebar-link
        v-for="(item, key) in items"
        :key="key"
        :item="item"
        :space="0"
      />
    </ul>
  </aside>
</template>

<script>
import AppSidebarLink from "@/components/app-sidebar-link/AppSidebarLink";


export default {
  name: "app-sidebar",
  components: {
    AppSidebarLink,
  },
  props: {
    items: Array,
  },

};
</script>

<style lang="scss">
@import "styles";
</style>
