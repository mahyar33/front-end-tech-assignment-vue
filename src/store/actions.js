import measurements from '@/assets/measurements.json'
import data from '@/assets/assets.json'
import {getMeasurements} from "@/utils/general";

export default {
    setCurrentAsset({commit}, payload) {
        commit('setCurrentAsset', payload)
    },
    setIsAggregated({commit},payload){
        commit('setIsAggregated', payload)
    },
    fetchMeasurements({getters,commit}, payload) {
        const result= getMeasurements(getters.getAssets, payload,measurements)
       commit('setMeasurements', result)
    },
    fetchAssets({commit}) {

        const idMapping = data.reduce((acc, el, i) => {
            acc[el.id] = i;
            return acc;
        }, {});
        let roots = [];
        data.forEach(el => {
            if (el.parentId === null) {
                delete el.parentId
                roots.push(el);
                return;
            }
            const parentEl = data[idMapping[el.parentId]];
            delete el.parentId
            parentEl.children = [...(parentEl.children || []), el];

        });

        commit('setAssets', roots)
    }
}