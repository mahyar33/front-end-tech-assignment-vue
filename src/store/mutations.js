export default {
    setCurrentAsset(state,payload){
        state.currentAsset=payload
    },
    setAssets(state,payload){
        state.assets=payload
    },
    setMeasurements(state,payload){
        state.measurements=payload
    },
    setIsAggregated(state,payload){
        state.isAggregated=payload
    }
}