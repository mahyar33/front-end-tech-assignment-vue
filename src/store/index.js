import { createStore } from "vuex";
import actions from './actions'
import mutations from './mutations'


export const defaultStore = {
  state: {
    currentAsset:'',
    assets:[],
    measurements:[],
    isAggregated:false
  },
  mutations,
  getters: {
    getCurrentAsset(state){
      return state.currentAsset
    },
    getAssets(state){
      return state.assets
    },
    getMeasurements(state){
      return state.measurements
    },
    getIsAggregated(state){
      return state.isAggregated
    }
  },
  actions,
}

export default createStore(defaultStore);
