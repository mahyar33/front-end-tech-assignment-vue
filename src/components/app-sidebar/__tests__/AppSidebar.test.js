import AppSidebar from '../AppSidebar';
import {render} from '@testing-library/vue';
import store from '../../../store/index'


describe('<app-sdiber-link />', () => {

    it('should render an <aside> tag', () => {
        const appSidebar = render(AppSidebar);
        expect(appSidebar.container.querySelector('aside')).toBeInTheDocument();
    });
    it('should match snapshot', () => {
        const appSidebar = render(AppSidebar);
        expect(appSidebar.container.firstChild).toMatchSnapshot();
    });
    it('should render one item', () => {
        const appSidebar = render(AppSidebar, {
            props: {
                items: [{
                    "id": 0,
                    "name": "Asset 0"
                }]
            },
            global: {
                plugins: [store]
            }

        });
        expect(appSidebar.container.querySelectorAll('li').length).toBe(1);
    });
});