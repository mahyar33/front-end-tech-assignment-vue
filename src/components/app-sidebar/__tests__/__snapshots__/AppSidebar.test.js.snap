// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<app-sdiber-link /> should match snapshot 1`] = `
<aside
  class="app-sidebar"
>
  <ul
    class="app-sidebar__menu"
  >
    
    
  </ul>
</aside>
`;
